runestone/assess/js/mchoice.js

```diff
@@ -284,64 +273,109 @@ MultipleChoice.prototype.randomizeAnswers = function () {
     }
 };
 
-/*====================================
-=== Repopulation from localstorage ===
-====================================*/
+/*===================================
+=== Checking/loading from storage ===
+===================================*/
 
-MultipleChoice.prototype.restoreLocalAnswers = function () {         // Handles local storage
-    if (this.multipleanswers) {
-        this.restoreMultipleSelect();
+MultipleChoice.prototype.checkServer = function () {
+    // Check if the server has stored answer
+    if (this.useRunestoneServices) {
+        var data = {};
+        data.div_id = this.divid;
+        data.course = eBookConfig.course;
+        data.event = "mChoice";
+        jQuery.get(eBookConfig.ajaxURL + "getAssessResults", data, this.repopulateFromStorage.bind(this)).error(this.checkLocalStorage.bind(this));
     } else {
-        this.restoreRadio();
+        this.checkLocalStorage();   // just go right to local storage
     }
-};
 
-MultipleChoice.prototype.restoreMultipleSelect = function () {
-    // This function repopulates MCMA questions with a user"s previous answers,
-    // which were stored into local storage.
-    var len = localStorage.length;
-    if (len > 0) {
+};
 
-        var ex = localStorage.getItem(eBookConfig.email + ":" + this.divid);
-        if (ex !== null) {
-            var arr = ex.split(";");
-            var answers = arr[0].split(",");
+MultipleChoice.prototype.repopulateFromStorage = function (data, status, whatever) {
+    if (data !== "") {
+        var dataEval = JSON.parse(data);
```

Comment:
This is strange, because in my setup data is already pre-parsed when it comes from a jquery request.  data has a timestamp an answer and a correct attribute.  Trying to JSON.parse it causes an error.